ddp-workspace/projects/toolkit/src/lib/components/dashboard/dashboard-redesigned.component.ts

```diff
@@ -51,37 +80,118 @@ import { map, take, filter } from 'rxjs/operators';
+        protected router: Router,
         private _announcements: AnnouncementsServiceAgent,
         private userInvitation: UserInvitationServiceAgent,
         _participantsSearch: ParticipantsSearchServiceAgent,
-        @Inject('toolkit.toolkitConfig') public config: ToolkitConfigurationService) {
-        super(_router, _announcements, _participantsSearch, config);
+        private governedParticipantsAgent: GovernedParticipantsServiceAgent,
+        private userActivityServiceAgent: UserActivityServiceAgent,
+        private userProfile: UserProfileServiceAgent,
+        private translate: TranslateService,
+        @Inject('ddp.config') private config: ConfigurationService,
+        @Inject('toolkit.toolkitConfig') public toolkitConfig: ToolkitConfigurationService) {
+        super(router, _announcements, _participantsSearch, session, toolkitConfig);
     }
 
     public ngOnInit(): void {
         super.ngOnInit();
         this.headerConfig.setupDefaultHeader();
         !this.isAdmin && this.getInvitationId();
+
+        this.dashboardParticipants$ = this.getDashboardParticipants();
+    }
+
+    private getOperatorActivities(): Observable<Array<ActivityInstance> | null> {
+        return of(this.session.session.userGuid).pipe(
+            tap((userGuid) => {
+                this.session.setParticipant(userGuid);
+            }),
+            switchMap(() => this.userActivityServiceAgent.getActivities(of(this.config.studyGuid))));
+    }
+
+    private getDashboardParticipants(): Observable<DashboardParticipant[]> {
+        return combineLatest([
+            this.getOperatorActivities()
+                .pipe(
+                    tap((userActivities) => {
+                        this.operatorActivities = userActivities;
+                    }),
+                    switchMap((userActivities) => userActivities?.length ? this.userProfile.profile : of(null))),
+            this.governedParticipantsAgent.getGovernedStudyParticipants(this.config.studyGuid)
+        ]).pipe(map(([operatorParticipant, participants]) => {
+            const result: DashboardParticipant[] = participants.map((participant, i) => ({
+                userGuid: participant.userGuid,
+                label: (participant.userProfile.firstName || participant.userProfile.lastName)
+                    ? `${participant.userProfile.firstName} ${participant.userProfile.lastName}`
+                    : `${this.translate.instant('Toolkit.Dashboard.ChildLabel')}${i > 0 ? ' #' + (i + 1) : ''}`,
+            }));
+            if (operatorParticipant) {
+                result.unshift({
+                    userGuid: this.session.session.userGuid,
+                    label: (operatorParticipant.profile.firstName || operatorParticipant.profile.lastName)
+                        ? `${operatorParticipant.profile.firstName} ${operatorParticipant.profile.lastName}`
+                        : this.translate.instant('Toolkit.Dashboard.UserLabel'),
```

Comment:
Code blocks `lines 181-183` and `lines 188-190` (labels definitions) seems pretty similar. 
Does it make sense to extract them into a separate method ?

Reply:
that's similar, but has enough differences, I'd prefer to not combine these 2 together